nilearn/experimental/surface/tests/test_maskers.py

```diff
@@ -77,15 +89,24 @@ def test_transform_inverse_transform(shape, make_mini_img, assert_img_equal):
     assert_img_equal(img, unmasked_img)
 
 
+@pytest.mark.xfail(
+    reason="Will implement a different surface image inside the test."
+)
 @pytest.mark.parametrize("shape", [(), (1,), (3,), (3, 2)])
+@pytest.mark.parametrize("n_zeros_in_mask", [4, 6])
 def test_transform_inverse_transform_with_mask(
-    shape, make_mini_img, assert_img_equal, mini_mask
+    make_surface_img,
+    assert_img_equal,
+    make_surface_mask,
+    shape,
+    n_zeros_in_mask,
 ):
-    img = make_mini_img(shape)
-    masker = SurfaceMasker(mini_mask).fit(img)
+    img = make_surface_img(shape)
+    mask = make_surface_mask(n_zeros_in_mask)
+    masker = SurfaceMasker(mask).fit(img)
     masked_img = masker.transform(img)
-    assert masked_img.shape == (*shape, img.shape[-1] - 2)
-    assert np.array_equal(masked_img.ravel()[:7], [2, 3, 4, 20, 30, 40, 50.0])
+    assert masked_img.shape == (*shape, img.shape[-1] - n_zeros_in_mask)
+    assert np.array_equal(masked_img.ravel()[:7], [2, 3, 4, 20, 30, 40, 50])
```

Comment:
Just a note for myself:

We are comparing hard-coded array values here. 
So, as discussed with @Remi-Gau, it might be best to define new SurfaceImage with specific data instead of using the fixture.